Warnings with Clang 10

With Clang 10.0.0, the following warnings appear.

```
../single_include/nlohmann/json.hpp:14558:53: warning: implicit conversion changes signedness: 'const int' to 'unsigned long' [-Wsign-conversion]
        return buf + (2 + static_cast<size_t>(-n) + k);
                                                  ~ ^
../single_include/nlohmann/json.hpp:14895:39: warning: implicit conversion loses integer precision: 'decltype(__x.base() - __y.base())' (aka 'long') to 'int'
      [-Wshorten-64-to-32]
                        int index = i - val.m_value.binary->cbegin();
                            ~~~~~   ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../single_include/nlohmann/json.hpp:17727:15: note: in instantiation of member function 'nlohmann::detail::serializer<nlohmann::basic_json<std::map, std::vector,
      std::__1::basic_string<char>, bool, long long, unsigned long long, double, std::allocator, adl_serializer, std::__1::vector<unsigned char, std::__1::allocator<unsigned
      char> > > >::dump' requested here
            s.dump(*this, true, ensure_ascii, static_cast<unsigned int>(indent), serialize_binary);
              ^
../single_include/nlohmann/json.hpp:24020:20: note: in instantiation of member function 'nlohmann::basic_json<std::map, std::vector, std::__1::basic_string<char>, bool, long
      long, unsigned long long, double, std::allocator, adl_serializer, std::__1::vector<unsigned char, std::__1::allocator<unsigned char> > >::dump' requested here
        return h(j.dump());
                   ^
```

These lines were last changed in 0a8216890 (#1967) and 012c9665a (#1662).

--- a/include/nlohmann/detail/conversions/to_chars.hpp
+++ b/include/nlohmann/detail/conversions/to_chars.hpp
@@ -1006,7 +1006,7 @@ inline char* format_buffer(char* buf, int len, int decimal_exponent,
 
         std::memmove(buf + (static_cast<size_t>(n) + 1), buf + n, static_cast<size_t>(k) - static_cast<size_t>(n));
         buf[n] = '.';
-        return buf + (static_cast<size_t>(k) + 1);
+        return buf + (static_cast<size_t>(k) + 1U);
     }
 
     if (min_exp < n and n <= 0)
@@ -1018,7 +1018,7 @@ inline char* format_buffer(char* buf, int len, int decimal_exponent,
         buf[0] = '0';
         buf[1] = '.';
         std::memset(buf + 2, '0', static_cast<size_t>(-n));
-        return buf + (2 + static_cast<size_t>(-n) + k);
+        return buf + (2U + static_cast<size_t>(-n) + static_cast<size_t>(k));
     }
 
     if (k == 1)

--- a/include/nlohmann/detail/output/serializer.hpp
+++ b/include/nlohmann/detail/output/serializer.hpp
@@ -259,8 +259,7 @@ class serializer
                     {
                         dump_integer(*i);
                         o->write_character(',');
-                        int index = i - val.m_value.binary->cbegin();
-                        if (index % 16 == 0)
+                        if (std::distance(val.m_value.binary->cbegin(), i) % 16 == 0)
                         {
                             o->write_character('\n');
                         }

--- a/single_include/nlohmann/json.hpp
+++ b/single_include/nlohmann/json.hpp
@@ -14543,7 +14543,7 @@ inline char* format_buffer(char* buf, int len, int decimal_exponent,
 
         std::memmove(buf + (static_cast<size_t>(n) + 1), buf + n, static_cast<size_t>(k) - static_cast<size_t>(n));
         buf[n] = '.';
-        return buf + (static_cast<size_t>(k) + 1);
+        return buf + (static_cast<size_t>(k) + 1U);
     }
 
     if (min_exp < n and n <= 0)
@@ -14555,7 +14555,7 @@ inline char* format_buffer(char* buf, int len, int decimal_exponent,
         buf[0] = '0';
         buf[1] = '.';
         std::memset(buf + 2, '0', static_cast<size_t>(-n));
-        return buf + (2 + static_cast<size_t>(-n) + k);
+        return buf + (2U + static_cast<size_t>(-n) + static_cast<size_t>(k));
     }
 
     if (k == 1)
@@ -14892,8 +14892,7 @@ class serializer
                     {
                         dump_integer(*i);
                         o->write_character(',');
-                        int index = i - val.m_value.binary->cbegin();
-                        if (index % 16 == 0)
+                        if (std::distance(val.m_value.binary->cbegin(), i) % 16 == 0)
                         {
                             o->write_character('\n');
                         }

--- a/test/src/unit-deserialization.cpp
+++ b/test/src/unit-deserialization.cpp
@@ -76,7 +76,7 @@ struct SaxEventLogger : public nlohmann::json_sax<json>
         return true;
     }
 
-    bool binary(std::vector<std::uint8_t>& val)
+    bool binary(std::vector<std::uint8_t>& val) override
     {
         std::string binary_contents = "binary(";
         std::string comma_space = "";

